lib/recipient.rb

```diff
@@ -0,0 +1,61 @@
+require "dotenv"
+require "httparty"
+Dotenv.load
+
+class Recipient
+  class SlackApiError < StandardError; end
+
+  attr_accessor :slack_id, :name
+
+  def initialize(slack_id, name)
+    @slack_id = slack_id
+    @name = name
+  end
+
+  CHANNEL_URL = "https://slack.com/api/channels.list"
+  USER_URL = "https://slack.com/api/users.list"
+  POST_URL = "https://slack.com/api/chat.postMessage"
```

Comment:
Ideally the superclass would not need to know these details from the subclass. One of the benefits of inheritance is it allows you to add new subclasses without changing the parent class.

To do so, instead of using a constant for this, you could define a template method `list_url` which is overridden in the child classes to return the correct endpoint.

That would also allow you to get rid of the `if` statement in `self.get` below.